Mathlib/Data/Finset/Update.lean

```diff
@@ -0,0 +1,62 @@
+/-
+Copyright (c) 2023 Floris van Doorn. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Floris van Doorn
+-/
+import Mathlib.Data.Finset.Basic
+
+/-!
+# Update a function on a set of values
+
+This file defines `Function.updateFinset`, the operation that updates a function on a
+(finite) set of values.
+-/
+variable {ι : Sort _} {π : ι → Sort _} {x : ∀ i, π i}
+
+namespace Function
+
+/-- `updateFinset x s y` is the vector `x` with the coordinates in `s` changed to the values of `y`.
+
+We should generalize `updateFinset` to `SetLike` instead of restricting to `Finset`.
+However, at the moment `Finset` is not `SetLike`, so we cannot do this yet.
+-/
+def updateFinset (x : ∀ i, π i) (s : Finset ι) [DecidablePred (· ∈ s)]
```

Comment:
I assume there's a reason you didn't just use `[DecidableEq _]` here? If so, can you add it to the docstring?

Reply:
`DecidablePred` generalizes better to other `SetLike` things. But I was inconsistently using `DecidableEq` in the rest of the file, so I just use that everywhere.